vendor/bat-native-ledger/src/bat/ledger/internal/wallet_provider/transfer.h

```diff
@@ -0,0 +1,67 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_VENDOR_BAT_NATIVE_LEDGER_SRC_BAT_LEDGER_INTERNAL_WALLET_PROVIDER_TRANSFER_H_
+#define BRAVE_VENDOR_BAT_NATIVE_LEDGER_SRC_BAT_LEDGER_INTERNAL_WALLET_PROVIDER_TRANSFER_H_
+
+#include <string>
+
+#include "bat/ledger/ledger.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+namespace ledger {
+class LedgerImpl;
+
+namespace wallet_provider {
+
+class Transfer {
+ public:
+  explicit Transfer(LedgerImpl*);
+
+  virtual ~Transfer();
+
+  void Run(const std::string& contribution_id,
+           const std::string& destination,
+           double amount,
+           ledger::ResultCallback) const;
```

Comment:
You mean declarations, right? We can! I started removing them some time ago where they didn't add any additional info. I can stop doing that and add them back as we move forward.

Actually, I'd also have a suggestion on specifying `const` for by-value types in function parameters:
- in declarations, we shouldn't, since the caller shouldn't care if the implementation assigns to the value passed to the function
- in definitions, we should treat function parameters as we would local variables (since, in fact, they act as local variables)

E.g.:
```
void f(int i);

void f(const int i) {
  // i is not being assigned to here
}
```

Reply:
I think that's a reasonable suggestion. Personally, I don't think `const` for locals really adds much value. I think we use it because it makes us feel like we're "serious" programmers. ("Real programmers hate mutability!") But your suggestion seems like a good compromise to me.

Reply:
Yeah, things should have been designed the other way around:
```
int i = 0;         // should be const int i = 0; semantics
mutable int i = 0; // should be int i = 0; semantics
```
And I think we've even had the `mutable` keyword since C++98!